File: sample/src/main/java/com/gibbs/sample/SimpleGPlayerViewActivity.java
package com.gibbs.sample;

import android.content.Intent;
import android.os.Bundle;

import com.gibbs.gplayer.GPlayerView;
import com.gibbs.gplayer.source.MediaSource;

public class SimpleGPlayerViewActivity extends BaseActivity {
    private GPlayerView mVideoView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_simple_g_player);
        mVideoView = findViewById(R.id.gl_surface_view);
        Intent intent = getIntent();
        String url = intent.getStringExtra("url");
        mVideoView.setUrl(MediaSource.SOURCE_TYPE_FILE, url);
    }

    @Override
    protected void onResume() {
        super.onResume();
        if (mVideoView != null) {
            mVideoView.onResume();
            mVideoView.startPlay();
        }
    }

    @Override
    protected void onPause() {
        super.onPause();
        if (mVideoView != null) {
            mVideoView.onPause();
            mVideoView.stopPlay();
        }
    }
}

File: sample/src/main/java/com/gibbs/sample/SimpleGPlayerViewActivity.kt
package com.gibbs.sample

import android.os.Bundle
import com.gibbs.gplayer.GPlayer
import com.gibbs.gplayer.listener.OnPreparedListener
import kotlinx.android.synthetic.main.activity_simple_gplayer.gl_surface_view

class SimpleGPlayerViewActivity : BaseActivity(), OnPreparedListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_simple_gplayer)
        val url = intent.getStringExtra("url")
        val useMediaCodec = intent.getBooleanExtra("useMediaCodec", false)
        if (useMediaCodec) {
            gl_surface_view.setFlags(GPlayer.USE_MEDIA_CODEC)
        }
        gl_surface_view.setDataSource(url)
        gl_surface_view.setOnPreparedListener(this)
    }

    override fun onStart() {
        super.onStart()
        gl_surface_view.onResume()
        gl_surface_view.prepare()
    }

    override fun onStop() {
        super.onStop()
        gl_surface_view.onPause()
        gl_surface_view.stop()
    }

    override fun onPrepared() {
        gl_surface_view.start()
    }
}